databroker/core.py

```diff
@@ -0,0 +1,543 @@
+from __future__ import print_function
+import six  # noqa
+from collections import defaultdict
+from itertools import chain
+import pandas as pd
+import tzlocal
+import doct as doc
+from pims import FramesSequence, Frame
+from metadatastore.commands import NoRunStop, NoEventDescriptors
+from filestore import handlers, HandlerBase
+import logging
+import numbers
+
+# Toolz and CyToolz have identical APIs -- same test suite, docstrings.
+try:
+    from cytoolz.dicttoolz import merge
+except ImportError:
+    from toolz.dicttoolz import merge
+
+
+logger = logging.getLogger(__name__)
+
+
+def fill_event(fs, event, handler_registry=None, handler_overrides=None):
+    """
+    Populate events with externally stored data.
+
+    Parameters
+    ----------
+    fs : FileStoreRO
+    event : document
+    handler_registry : dict, optional
+        mapping spec names (strings) to handlers (callable classes)
+    handler_overrides : dict, optional
+        mapping data keys (strings) to handlers (callable classes)
+    """
+    if handler_overrides is None:
+        handler_overrides = {}
+    is_external = _inspect_descriptor(event.descriptor)
+    mock_registries = {data_key: defaultdict(lambda: handler)
+                       for data_key, handler in handler_overrides.items()}
+    for data_key, value in six.iteritems(event.data):
+        if is_external.get(data_key, False):
+            if data_key not in handler_overrides:
```

Comment:
Probably should combine these two paths into one since only the registry differs